### ISSUE TITLE
Ctrl-W and Ctrl-M

### ISSUE THREAD

**Issue by Chinggis6** [zeit/hyper#2050]
Please tell me a way to disable those shortcuts as they mess with readline keybindings when I `ssh` into a Linux machine.

`Ctrl-W` is for removing a word back
`Ctrl-M` is for <kbd>Enter</kbd>

They are especially useful when one uses Readline with Vim key bindings.

---

I can disable them editing `accelerators.js`  file so now the window doesn't get closed or minimized, but they still block those keys for Readline to act on.

I want to be able to use Readline with those shortcuts. I don't want Hyper to block them for me.

**Comment by waretyler** [comment]
This should be fixed if you build from master. I don't have anywhere I can ssh to right now, so I can't fully verify, but I'm on a linux box. `Ctrl-w` and `Ctrl-m` work as expected (this doesn't work for me in the released apps only if I build from master). Seems related to (fixed by) this pull request:  https://github.com/zeit/hyper/pull/1876

**Comment by ppot** [comment]
@Chinggis6 If you can test the laster master it would be great. If you have any problem feel free to tag me with a specific description!

**Comment by thisguychris** [comment]
I'm gonna add `Ctrl-N` here, use for vim's completion, it opens a new window.

**Comment by Chinggis6** [comment]
I have removed and reinstalled Hyper from Releases subpage (v1.3.3).
The problem is not gone, `Ctrl-M` still minimizes and `Ctrl-W` closes the window.

I tried to clone and build using `yarn` what I downloaded and installed separately (directly from `yarn-0.27.5.msi`).

`yarn run dev` - `webpack -w` is not recognized as a command (not found)

`yarn run dist` -  `cross-env` is not recognized as a command

I also tried `npm i -g webpack` and now it says cannot find module `webpack`.

**I have limited `yarn/nodejs` experience**

It would be very cool if everything got solved by `cinst hyper` or better `cup hyper` which are shorter aliases for `choco install` and `choco upgrade` respectively.

**Comment by ppot** [comment]
In the current state, `master` is not a release.

**Comment by Chinggis6** [comment]
I know, any tips how to build Hyper from scratch on Windows?

**Comment by ppot** [comment]
Head to https://github.com/zeit/hyper and go on app-vayor icon build. Than download the artifact. Tell me if it work

**Comment by waretyler** [comment]
@Chinggis6 Probably try @ppot's suggestion first, but if you want to know how to get it building I wrote up a response to your initial question. 

---

I'm not a windows box, but I think that shouldn't impact the steps too much. 
First you'll want to make sure you've added the libraries listed in ['Contribute' section](https://github.com/zeit/hyper#contribute) of the readme. 

For node based / js oriented projects there is typically a `package.json` file that contains a lot of 
information about the projects dependencies and useful scripts for the project. (I mention this in case you want to dig a little into how it is working)

First we should start with a clean slate. Make sue you've pulled the latest code then run `yarn run clean` (this will clean up untracked directories). Then run the command `yarn` - this will install the node modules in the project root and the `app` sub-directory. Then give `yarn run dist` a try, it should install all the binaries & libraries you need to build in `node_modules` and `app/node_modules`. 

If that doesn't work I'm happy to help you dig into it more. Just let us know what errors you're seeing.

**Comment by Chinggis6** [comment]
```
gyp ERR! System Windows_NT 10.0.15063
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Administrator\hh\hyper\app\node_modules\node-pty
gyp ERR! node -v v8.1.3
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok

    at ChildProcess.childProcess.once.code (C:\Users\Administrator\hh\hyper\node_modules\electron-builder-util\src\util.ts:141:14)
    at Object.onceWrapper (events.js:318:30)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:897:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)
From previous event:
    at spawn (C:\Users\Administrator\hh\hyper\node_modules\electron-builder-util\src\util.ts:104:3)
    at installDependencies (C:\Users\Administrator\hh\hyper\node_modules\electron-builder\src\util\yarn.ts:87:3)
    at C:\Users\Administrator\hh\hyper\node_modules\electron-builder\src\util\yarn.ts:17:11
    at Generator.next (<anonymous>)
From previous event:
    at installOrRebuild (C:\Users\Administrator\hh\hyper\node_modules\electron-builder\out\util\yarn.js:34:21)
    at C:\Users\Administrator\hh\hyper\node_modules\electron-builder\src\cli\install-app-deps.ts:52:42
    at Generator.next (<anonymous>)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
From previous event:
    at installAppDeps (C:\Users\Administrator\hh\hyper\node_modules\electron-builder\out\cli\install-app-deps.js:48:21)
    at Object.args [as handler] (C:\Users\Administrator\hh\hyper\node_modules\electron-builder\out\cli\cli.js:121:9)
    at Object.self.runCommand (C:\Users\Administrator\hh\hyper\node_modules\yargs\lib\command.js:233:22)
    at Object.Yargs.self._parseArgs (C:\Users\Administrator\hh\hyper\node_modules\yargs\yargs.js:990:30)
    at Object.get [as argv] (C:\Users\Administrator\hh\hyper\node_modules\yargs\yargs.js:927:19)
    at Object.<anonymous> (C:\Users\Administrator\hh\hyper\node_modules\electron-builder\out\cli\cli.js:117:465)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3
error Command failed with exit code 4294967295.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

I get the above *error* after running `yarn`.
I use Administrator account under Windows 10.

**Comment by waretyler** [comment]
I wasn't familiar with appveyor until I got curious. There is a little icon at the top of the readme for it (no text to indicate, but you can hover over it). I'll short circuit your need to find it - here is the link to hyper's appveyor build artifacts: https://ci.appveyor.com/project/zeit/hyper/build/artifacts


---


If you still want to build:

Not seeing any smoking guns. I don't know how much help I will be, but I will keep giving you sort of random commands to run (ones that I would if I were at your computer)

Options:

a) This might be an issue with `node-pty`, which I saw listed on the readme. You can run `yarn run rebuild-node-pty` and then `yarn` to see if it comes out clean. It might be a good idea to run `yarn run clean` if it doesn't work the first time and try again. 

b) you can try the other commands first `yarn run dev` and `yarn run app` and see if those work

c) Ditch yarn for the dependency install and just install the dependencies manually with npm. First run `yarn run clean`. Then in the project root run `npm install`. Change directories to the app directory and run `npm install` then go back to project root and run `npm run dist`. This also _should_ be a viable way to build

d) you can also do a gut check with a simpler electron application such as https://github.com/electron/electron-quick-start, which has a pretty quick set up instructions. The benefit of this is you can verify if it's an issue with some subset of the underlying tools or hyper in particular.

**Comment by Chinggis6** [comment]
Thank you for the help so far.
I still get a lot of error running those commands.

Also, I downloaded and installed the build from AppVeyor, and now Hyper doesn't open up at all, though I see the process running in Task Manager.

Is it not possible modify the configuration to just unmap those keys without affecting Readline and without re-installing the application?

**Comment by ppot** [comment]
@Chinggis6 Unfortunately not, the keymap is one of the big thing integrated into `1.4.0` release.

**Comment by waretyler** [comment]
@Chinggis6 - I'll play around with building on my windows machine this weekend and see if I can get it working 🙂

**Comment by ppot** [comment]
Will close this since it's working in master. Fix in #1876 
If after release you still have an issue, feel free to create a new issue and tag me.
